### PR TITLE
LOOP-1486: add suspend threshold information screen + editor

### PR DESCRIPTION
**Pull request by novalegra**
https://tidepool.atlassian.net/browse/LOOP-1486

This PR also makes `TherapySettings` be a subcomponent of a `MockPrescription`

### CODE REVIEW

File: LoopKit/TherapySettings.swift [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 import HealthKit
 
-public struct TherapySettings: Equatable {
+public struct TherapySettings: Equatable, Codable {
```
Comment (novalegra) [inline LoopKit/TherapySettings.swift:11]: These changes are to enable `TherapySettings` to be a subcomponent of a prescription

File: LoopKitUI/Views/Settings Editors/SuspendThresholdEditor.swift [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+            }
+        )
+        .alert(isPresented: $showingConfirmationAlert, content: confirmationAlert)
+        .onTapGesture {
+            self.userDidTap = true
+        }
+    }
+
+    var description: Text {
+        Text("When your glucose is predicted to go below this value, the app will recommend a basal rate of 0 U/h and will not recommend a bolus.", comment: "Suspend threshold description")
```
Comment (nhamming) [inline LoopKitUI/Views/Settings Editors/SuspendThresholdEditor.swift]: [nit] elsewhere `LocalizedString` is being used.

**Review by nhamming — APPROVED** [review]
LGTM

File: LoopKit/TherapySettings.swift [hunk 3]
```diff
@@ -26,15 +26,21 @@ public struct TherapySettings: Equatable {
...
         maximumBolus: Double? = nil,
         suspendThreshold: GlucoseThreshold? = nil,
         insulinSensitivitySchedule: InsulinSensitivitySchedule? = nil,
-        carbRatioSchedule: CarbRatioSchedule? = nil
+        carbRatioSchedule: CarbRatioSchedule? = nil,
+        basalRateSchedule: BasalRateSchedule? = nil,
+        insulinModel: StoredSettings.InsulinModel? = nil
     ){
         self.glucoseTargetRangeSchedule = glucoseTargetRangeSchedule
         self.preMealTargetRange = preMealTargetRange
```
Comment (rickpasetto) [inline LoopKit/TherapySettings.swift:46]: I had fixed this in my branch: workoutTargetRange is never initialized. O_o.  (I realize this is somewhat orthogonal to _this_ PR)
```suggestion
        self.preMealTargetRange = preMealTargetRange
        self.workoutTargetRange = workoutTargetRange
```

**Review by rickpasetto — CHANGES_REQUESTED** [review]
I have one requested fix...but other than that LGTM!

**Comment by novalegra** [comment]
> I have one requested fix...but other than that LGTM!

Done - thanks for catching that!

**Review by rickpasetto — APPROVED** [review]
LGTM

**Pull request merged**